ui-tests/tools/com.liferay.ide.ui.jsf/com.liferay.ide.ui.jsf.wizard/com.liferay.ide.ui.jsf.wizard.validation/src/com/liferay/ide/ui/jsf/tests/ValidationJsfProjectTests.java

```diff
@@ -0,0 +1,188 @@
+
+	@Test
+	public void checkInitialState() {
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		validationAction.assertEquals(PLEASE_ENTER_A_PROJECT_NAME, wizardAction.getValidationMsg(2));
+
+		validationAction.assertTextEquals(StringPool.BLANK, wizardAction.newLiferayJsf.projectName());
+
+		validationAction.assertCheckedTrue(wizardAction.newLiferayJsf.useDefaultLocation());
+
+		validationAction.assertEnabledFalse(wizardAction.getFinishBtn());
+
+		validationAction.assertEnabledTrue(wizardAction.getCancelBtn());
+
+		wizardAction.cancel();
+	}
+
+	@Test
+	public void checkLocationWithWorkspace() {
+		String exceptLocation = envAction.getEclipseWorkspacePathOSString();
+
+		if ("win32".equals(Platform.getOS())) {
+			exceptLocation = exceptLocation.replaceAll("\\\\", "/");
+		}
+
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		wizardAction.newLiferayJsf.prepareGradle(project.getName());
+
+		ide.sleep();
+
+		validationAction.assertEquals(exceptLocation, wizardAction.newLiferayJsf.getLocation());
+
+		wizardAction.cancel();
+
+		wizardAction.openNewLiferayWorkspaceWizard();
+
+		wizardAction.newLiferayWorkspace.prepareGradle(projects.getName(0));
+
+		wizardAction.finish();
+
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		wizardAction.newLiferayJsf.prepareGradle(projects.getName(1));
+
+		ide.sleep();
+
+		validationAction.assertEquals(exceptLocation, wizardAction.newLiferayJsf.getLocation());
+
+		wizardAction.cancel();
+
+		viewAction.project.closeAndDelete(projects.getName(0));
+	}
+
+	@Test
+	public void checkWaitGradleBackgroundJobMessage() {
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		wizardAction.newLiferayJsf.prepareGradle(projects.getName(0), ICEFACES);
+
+		wizardAction.finish();
+
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		wizardAction.newLiferayJsf.prepareGradle(projects.getName(1), ICEFACES);
+
+		validationAction.assertEquals(
+			PROJECT_WONT_BE_RESOLVED_COMPLETELY_UNTIL_ALL_GRADLE_BACKGROUND_JOBS_FINISH,
+			wizardAction.getValidationMsg(2));
+
+		wizardAction.cancel();
+
+		jobAction.waitForNoRunningJobs();
+
+		viewAction.project.closeAndDelete(projects.getName(0));
+	}
+
+	@Test
+	public void validateBuildType() {
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		String[] expectedBuildTypes = {GRADLE, MAVEN};
+		String[] buildTypes = wizardAction.newLiferayJsf.buildTypes();
+
+		validationAction.assertLengthEquals(expectedBuildTypes, buildTypes);
+
+		for (int i = 0; i < buildTypes.length; i++) {
+			validationAction.assertEquals(expectedBuildTypes[i], buildTypes[i]);
+		}
+
+		wizardAction.cancel();
+	}
+
+	@Test
+	public void validateJsfProjectComponentSuite() {
+		String[] templates = {LIFERAY_FACES_ALLOY, ICEFACES, JSF_STANDARD, PRIMEFACES, RICHFACES};
+
+		wizardAction.openNewLiferayJsfProjectWizard();
+
+		wizardAction.newLiferayJsf.prepareGradle(project.getName());
+
+		validationAction.assertTextEquals(JSF_STANDARD, wizardAction.newLiferayJsf.projectComponentSuite());
```

Comment:
move check jsf_standard to initialState(). It will always be jsf_standard by default.